src/lib/apmJson.js

```diff
@@ -0,0 +1,186 @@
+   * @param {string} instPath - An installation path
+   * @param {string} keys - Keys to check existing
+   * @returns {Promise<boolean>} Whether `apm.json` has the property.
+   */
+  has: function (instPath, keys) {
+    if (!keys) return false;
+
+    const keysArray = keys.split('.');
+
+    let object = getApmJson(instPath);
+    for (const [i, key] of keysArray.entries()) {
+      if (Object.prototype.hasOwnProperty.call(object, key)) {
+        object = object[key];
+        if (object === null) {
+          if (i !== keysArray.length - 1) {
+            return false;
+          }
+          break;
+        }
+      } else {
+        return false;
+      }
+    }
+    return true;
+  },
+
+  /**
+   * Gets the value from `apm.json`.
+   *
+   * @param {string} instPath - An installation path
+   * @param {string} keys - Keys to get value
+   * @param {any} defaultValue - A value replaced when the property don't exists.
+   * @returns {Promise<any>} The property selected by key.
+   */
+  get: function (instPath, keys = '', defaultValue = undefined) {
+    const apmJson = getApmJson(instPath);
+
+    if (keys === '') return apmJson;
+    const keyArray = keys.split('.');
+
+    let object = apmJson;
+    for (const [i, key] of keyArray.entries()) {
+      object = object[key];
+
+      if (object === undefined) {
+        break;
+      } else if (object === null) {
+        if (i !== keyArray.length - 1) {
+          object = undefined;
+        }
+        break;
+      }
+    }
+    return object === undefined ? defaultValue : object;
+  },
+
+  /**
+   * Sets the value to `apm.json`.
+   *
+   * @param {string} instPath - An installation path
+   * @param {string} keys - Keys to set value
+   * @param {any} value - A value to set
+   */
+  set: function (instPath, keys, value) {
+    if (!keys) return;
+
+    const keyArray = keys.split('.');
+
+    let object = getApmJson(instPath);
+    for (const [i, key] of keyArray.entries()) {
+      if (i !== keyArray.length - 1) {
+        if (!(typeof object === 'object')) {
+          object[key] = {};
+        }
+        object = object[key];
+      } else {
+        object[key] = value;
+      }
+    }
+  },
+
+  /**
+   * Sets the core version to `apm.json`.
+   *
+   * @param {string} instPath - An installation path
+   * @param {string} program - A name of the program
+   * @param {string} version - A version of the program
+   */
+  setCore: function (instPath, program, version) {
+    const apmJson = getApmJson(instPath);
+    apmJson.core[program] = version;
+    setApmJson(instPath, apmJson);
+  },
+
+  /**
+   * Adds the information of the plugin to `apm.json`.
+   *
+   * @param {string} instPath - An installation path
+   * @param {object} plugin - An information of a plugin
+   */
+  addPlugin: function (instPath, plugin) {
+    const apmJson = getApmJson(instPath);
+    apmJson.plugins[plugin.id] = {
+      repository: plugin.repo,
+      version: plugin.latestVersion,
+    };
```

Comment:
```suggestion
    apmJson.plugins[plugin.repo + plugin.id] = {
      id: plugin.id,
      repository: plugin.repo,
      version: plugin.latestVersion,
    };
```

Reply:
Do you think this kind of change is acceptable?

```suggestion
    apmJson.plugins[plugin.repo][plugin.id] = {
      id: plugin.id,
      repository: plugin.repo,
      version: plugin.latestVersion,
    };
```

Reply:
Your way is better.

Reply:
Please wait a moment.
I think there is only one plugin/script that can be installed in the same directory but can duplicate IDs happen?

Reply:
Sorry, the duplication only occurs when reverse lookup is done from the installation information, so as you pointed out, there is no need to take action here.

Reply:
It's okay.
So, for the convenience, I will just add the id property.

```suggestion
    apmJson.plugins[plugin.id] = {
      id: plugin.id,
      repository: plugin.repo,
      version: plugin.latestVersion,
    };
```